code/src/renderer/components/File.js

```diff
@@ -123,6 +128,62 @@ export default class File extends React.Component {
     this.outputRef.current.forceOpen();
   };
 
+  test = async () => {
+    if (this.state.location) {
+      await this.save();
+    }
+    if (this.identifyLanguage() !== PYTHON) {
+      send({
+        type: SHOW_ERROR_DIALOG,
+        title: "Unable to Test",
+        message: "Doctests are only implemented for Python at the moment",
+      });
+      return;
+    }
+    let commandSent = false;
+    const [interactCallback, killCallback, detachCallback] = runCode(
+      this.identifyLanguage()
+    )(
+      this.state.editorText,
+      (out) => {
+        if (!out.startsWith(DOCTEST_MARKER)) {
+          return;
+        }
+        const rawData = out.slice(DOCTEST_MARKER.length);
+        // eslint-disable-next-line no-eval
+        const doctestData = (0, eval)(rawData);
```

Comment:
yeah, I hate JavaScript, but we need this because Python strings repr using single quotes, which JSON.parse hates, and my old hack to work around this isn't actually correct